src/algebra/gcd_monoid/basic.lean

```diff
@@ -806,6 +806,12 @@ end⟩
 
 @[simp] lemma normalize_eq (x : α) : normalize x = x := mul_one x
 
+@[simp] lemma lcm_mul_pow_eq_left [normalized_gcd_monoid α] {x y : α} (m : ℕ) [fact (0 < m)] :
+  lcm (x * y^m) y = x * y^m :=
+(lcm_eq_left_iff _ _ (by {convert normalize_eq (x * y^m)})).2
+  (dvd_mul_of_dvd_right (dvd_pow_self y (ne_zero_of_lt' 0)) x)
+-- normalize from different places, so convert needed
```

Comment:
Are you sure you want this in the `unique_units` section of the file? I think if you put it in the right palce then the need for this `convert` goes away.

Reply:
This proof doesn't need `unique (units R)` right? The right place is somewhere without that assumption.

Reply:
Ah, if there's a diamond here then `normalize_eq` is missing a`[normalization_monoid α]` argument, and the convert belongs there

Reply:
I'm not sure I understand, `normalize_eq` is based on `norm_unit_eq_one`, which is true for the specific structure given to `unique (units R)`..

Reply:
Yeah, the diamond is that under the assumptions (`[cancel_comm_monoid_with_zero α]`, `[unique αˣ]`, `[normalized_gcd_monoid α]`) there are two distinct ways to make a `normalization_monoid`; one via `normalization_monoid_of_unique_units` (which uses the first two assumptions) and one via `normalized_gcd_monoid.to_normalization_monoid` (which uses the first and third).

Reply:
If you add an instance `gcd_monoid` + `unique units` => `normalized_gcd_monoid` then the assumption of the lemma can be weakened to `gcd_monoid` and the problem goes away. I pushed a fix.